debezium-core/src/main/java/io/debezium/relational/history/JdbcDatabaseHistory.java

```diff
@@ -0,0 +1,241 @@
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+import org.apache.kafka.connect.errors.ConnectException;
+
+import com.google.common.base.Splitter;
+
+import io.debezium.annotation.ThreadSafe;
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.document.DocumentReader;
+import io.debezium.document.DocumentWriter;
+import io.debezium.util.Collect;
+import io.debezium.util.FunctionalReadWriteLock;
+
+/**
+ * A {@link DatabaseHistory} implementation that stores the schema history to database table
+ *
+ * @author Ismail Simsek
+ */
+@ThreadSafe
+public final class JdbcDatabaseHistory extends AbstractDatabaseHistory {
+
+    public static final Field JDBC_URI = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "jdbc.uri")
+            .withDescription("Uri of the database which will be used to record the database history")
+            .withValidation(Field::isRequired);
+    public static final Field JDBC_USER = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "jdbc.user")
+            .withDescription("Username of the database which will be used to record the database history")
+            .withValidation(Field::isRequired);
+    public static final Field JDBC_PASSWORD = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "jdbc.password")
+            .withDescription("Password of the database which will be used to record the database history")
+            .withValidation(Field::isRequired);
+    public static final String DATABASE_HISTORY_TABLE_NAME = "debezium_database_history";
+    public static final String DATABASE_HISTORY_TABLE_DDL = "CREATE TABLE " + DATABASE_HISTORY_TABLE_NAME +
+            "(" +
+            "id VARCHAR(36) NOT NULL," +
+            "history_data VARCHAR(65000)," +
+            "history_data_seq INTEGER," +
+            "record_insert_ts TIMESTAMP NOT NULL," +
+            "record_insert_seq INTEGER NOT NULL" +
+            ")";
+    public static final String DATABASE_HISTORY_TABLE_SELECT = "SELECT id, history_data, history_data_seq FROM " + DATABASE_HISTORY_TABLE_NAME
+            + " ORDER BY record_insert_ts, record_insert_seq, id, history_data_seq";
+    public static final String DATABASE_HISTORY_TABLE_INSERT = "INSERT INTO " + DATABASE_HISTORY_TABLE_NAME + " VALUES ( ?, ?, ?, ?, ? )";
+    public static Collection<Field> ALL_FIELDS = Collect.arrayListOf(JDBC_USER);
+    private final FunctionalReadWriteLock lock = FunctionalReadWriteLock.reentrant();
+    private final DocumentWriter writer = DocumentWriter.defaultWriter();
+    private final DocumentReader reader = DocumentReader.defaultReader();
+    private final AtomicBoolean running = new AtomicBoolean();
+    AtomicInteger recordInsertSeq = new AtomicInteger(0);
```

Comment:
using AtomicInteger to guarantee uniqueness withing the jvm process, incase Timestamp values are same {'record_insert_ts' +' record_insert_seq'} combination should be unique

Reply:
changed to ` private final`